### DISCUSSION TITLE
File Upload using built in database

### DISCUSSION THREAD

**Discussion by yavuzaydn** [serverpod/serverpod#2093] [Q&A] [answered]
Hi, I have been trying to upload an image to the built in storage using ios simulator and image picker.
session.storage.getPublicUrl() gives the following error:
<img width="762" alt="Ekran Resmi 2024-04-01 20 49 41" src="https://github.com/serverpod/serverpod/assets/88391056/8b081a0d-c22d-4747-a52b-f724ca144edb">
<img width="908" alt="Ekran Resmi 2024-04-01 20 49 54" src="https://github.com/serverpod/serverpod/assets/88391056/c766999b-0aa6-4d08-8321-5608907b1466">

here is my endpoint:
<img width="745" alt="Ekran Resmi 2024-04-01 20 59 33" src="https://github.com/serverpod/serverpod/assets/88391056/9123deae-e59c-4a43-a6bd-58e193a22761">

I can also see the data in serverpod_cloud_storage:
<img width="1051" alt="Ekran Resmi 2024-04-01 20 49 25" src="https://github.com/serverpod/serverpod/assets/88391056/11214cf3-2372-4245-a949-a26290ea6fca">
 I would appreciate any guidance you can provide.

**Comment by Isakdl** [comment, marked as answer]
Uri is currently not supported in the serialization (issue tracked here: https://github.com/serverpod/serverpod/issues/1784). If you convert the URI to a string before sending it, it should work.

A word of caution on the implementation here, I would recommend setting the path from the server and not letting the client decide the path. You can easily get into a situation with conflicting paths, or worse expose your backend to nefarious actors who can put files in locations you may not expect. This is especially important if you change the storage solution from the database to a bucket or similar.

**Reply by yavuzaydn** [reply]
Thank you @Isakdl. I successfully uploaded an image whose size is less than 1024 bytes but I cannot send an image larger than 1024 bytes. The verify upload method returns false. Do you know what can be the reason?

**Reply by estevez-dev** [reply]
Hi @yavuzaydn. I have similar issue: https://github.com/serverpod/serverpod/discussions/2617 . Did you find the reason?

**Reply by estevez-dev** [reply]
@yavuzaydn, found it https://github.com/serverpod/serverpod/issues/141
